### DISCUSSION TITLE
Realtime API lost connection

### DISCUSSION THREAD

**Discussion by Oggy4th** [pocketbase/pocketbase#2323] [Q&A]
Realtime API suddenly lost connection after few minutes and gives no error. Couldn't reconnect unless restart.

I don't know if its digital ocean's problem or SDK's (both Dart and JS).

Is there a way to configure keep-alive duration?

**Comment by ganigeorgiev** [comment]
The realtime api have 5 mins idle timeout, aka. if there was no event message sent during this time the client will be disconnected to avoid leaked/"zombie" connections.

This shouldn't be an issue with the JS or Dart SDK (or any other SSE client) since there is a automatic reconnect that will take care for establishing a new connection.

If that's not the case, please elaborate and provide a little more context about your setup.

**Reply by Oggy4th** [reply]
Hello there, @ganigeorgiev Thanks for the quick responses

I am using .subscribe() in Dart. Do I need to write reconnect logic or is it already in the SDK?

```
// Subscribe to changes only in the specified record
pb.collection('users').subscribe('RECORD_ID', (e) {
    console.log(e.record);
});
```

**Reply by ganigeorgiev** [reply]
> Do I need to write reconnect logic or is it already in the SDK?

No, the reconnect is built-in in the SDK, users don't have to do anything in addition.

Note that if you are testing with Flutter Web the realtime services at the moment are not supported there. This and other limitations you can check in https://github.com/pocketbase/dart-sdk#limitations.

**Reply by ganigeorgiev** [reply]
If PocketBase is deployed behind nginx or other reverse proxy, you may want to check the keepalive note in https://pocketbase.io/docs/going-to-production/#using-reverse-proxy.

**Reply by Oggy4th** [reply]
Built-in reconnect doesn't seem to work in flutter desktop. I'll try to produce minimal reproduction code.

**Reply by Oggy4th** [reply]
> If PocketBase is deployed behind nginx or other reverse proxy, you may want to check the keepalive note in https://pocketbase.io/docs/going-to-production/#using-reverse-proxy.

It's hosted in Digital Ocean without any reverse proxy.

**Reply by ganigeorgiev** [reply]
What version of the SDK you are using. It could be related to the latest OAuth2 realtime changes in v0.9.0. If you are using v0.9, could you try with v0.8? I'll try to test sometime later today.

**Reply by Oggy4th** [reply]
I was using v0.8. let you know after try v0.9

**Reply by ganigeorgiev** [reply]
There is a minor bug in v0.9 introduced with the new `Completer` but v0.8 should be fine. I'm not able to reproduce the issue locally. I'll submit a fix for the v0.9 sometime later today.

**Reply by ganigeorgiev** [reply]
The v0.9.0 `Completer` bug should be fixed in [Dart SDK v0.9.1](https://pub.dev/packages/pocketbase/changelog).

I've tested it locally with the template Flutter app on desktop (`flutter create myapp`) by subscribing to a collection and manually restarting the PocketBase server to ensure that the reconnect is executed.

If you still have issues, please create a minimal reproducible repo and I'll try to investigate it further.

**Reply by kisinga** [reply]
I also get a similar error:
"Connection was reset" after a few minutes of idle time. I'm using 0.18.0 version of the SDK, and it doesn't reconnect.

@ganigeorgiev  By reconnect do you mean that it should maintain the subscription?

**Reply by ganigeorgiev** [reply]
@kisinga I'm not sure that I understand what do you mean. Please open a new discussion with more details about your project (_note that the latest Dart SDK version is v0.15.1; I guess v0.18.0 is the version of your PocketBase executable_) and what platform are you targeting.

If you are testing on Flutter Web, please check the limitations section at https://github.com/pocketbase/dart-sdk#limitations.

Also note if you are using the `fetch_client` there is currently a "bug" with the dartjs interop in Chrome 117+ - https://github.com/dart-lang/sdk/issues/53532.
